src/librustc/traits/util.rs

```diff
@@ -137,7 +155,8 @@ impl Elaborator<'tcx> {
                 // cases. One common case is when people define
```

Comment:
So can we hit the infinite recursion now? e.g. say with `trait A: B + A {}` (or reversed order), `trait B {}` and then `Box<dyn A>` ~> `Box<dyn B>`. Can you add ^-- as tests perhaps?

Reply:
Correct: the trait graph should be guaranteed acyclic by this point.

Reply:
Ah 🎉 Just to be on the safe side, having the tests would still be great. :)

Reply:
Yep, no problem.